src/utils/hooks/youtubeAPI.js

```diff
@@ -0,0 +1,17 @@
+import { API_KEY } from "../constants";
+
+
+const fetchVideosFromQuery = (query) => {
+  const sanitizedQuery = encodeURI(query);
+  const url = `https://youtube.googleapis.com/youtube/v3/search?part=snippet&maxResults=25&q=${sanitizedQuery}&type=video&videoType=any&key=${API_KEY}`;
+  return fetch(url);
+};
+
+const fetchRelatedVideosFromId = (videoId) => {
+  const url = `https://youtube.googleapis.com/youtube/v3/search?part=snippet&maxResults=10&relatedToVideoId=${videoId}&type=video&key=${API_KEY}`;
+  return fetch(url);
+};
+
+const fetchAPI = { fetchVideosFromQuery, fetchRelatedVideosFromId };
```

Comment:
it's not a hook at all. You can create a custom hook to fetch the data and output the data

Reply:
I now have a better understanding of custom hooks, sorry about that